src/Mod/Sketcher/Gui/ViewProviderSketch.h

```diff
@@ -535,6 +542,7 @@ class SketcherGuiExport ViewProviderSketch : public PartGui::ViewProvider2DObjec
     /// signals if the elements list has changed
     boost::signals2::signal<void ()> signalElementsChanged;
     //@}
+    TaskSketcherTool* toolSettings;

```

Comment:
Strong code smell. Why does the ViewProvider need to store a pointer to a dialog in the taskbar? Worse, why does it need to expose it via its public interface to anybody with a pointer?

It appears that the VP does not need this information and it is only here as a convenient container.

If you need to signal something within the responsibility of the VP to a dialog (for example in the taskbar), you may use a boost::signals2 (see signalSetup in ViewProviderSketch for an example). But it does not appear to be here the case.

Reply:
It is totally as a convenient container. It's because I need to access the dialog from the DrawSketchHandler.

Reply:
Then, the smell was right. Refer to the solution above to get the access you need.